### ISSUE TITLE
convert to python 3

### ISSUE THREAD

**Issue by chessybo** [SiEPIC/SiEPIClab#5]
has any thought been given to convert this to python 3? Having it in python 2 is a bit inconvenient for most users that use python 3. 
https://stackoverflow.com/questions/42788383/switch-between-spyder-for-python-2-and-3/42788793

I read that the 2to3 module should be able to accomplish this, however, python.org says, "The lib2to3 API should be considered unstable"

**Comment by StephenHLin** [comment]
Hi chessybo, thanks for the interest.
We have discussed in our lab this a few times but never in detail. In theory the conversion for the majority of the scripts should be pretty easy (just print statements need to be changed to include parenthesis I believe). The only challenge may be the `ctypes` library and how to port over DLLs.

However, given the COVID situation for the last year or so, our team is focusing on using an in-house lightlab solution instead (https://lightlab.readthedocs.io/en/master/). In the long run, lightlab will also allow for remote control by collaborative partners and would be very handy for conferences. e.g. demonstrations via remote access, no need to bring hardware anymore.

**Comment by chessybo** [comment]
well, i've started the process myself. trying to do it in spyder, since when i try to open with pythonw nothing happens. 
I pip installed 2to3
I went to the package directory in Powershell and went 
`2to3 . -w`
Then had to do it again for pyOptomip.pyw for some reason
`2to3 pyOptomip.pyw -w`
Then had to pip install pyvisa-py

now when I run pyOptomip.pyw in Spyder I get the error, 
`
wxAssertionError: C++ assertion "!(flags & wxALIGN_RIGHT)" failed at ..\..\src\common\sizer.cpp(2168) in wxBoxSizer::DoInsert(): Horizontal alignment flags are ignored in horizontal sizers`

I'm unfamiliar with wx, the traceback is,

```

  File "C:\ProgramData\Miniconda3\Lib\site-packages\pyOptomip\CorvusEcoParameters.py", line 48, in InitUI
    self.para1.AddMany([(self.para1name,1,wx.EXPAND|wx.ALIGN_LEFT),(self.para1tc,1,wx.EXPAND|wx.ALIGN_RIGHT)])

  File "C:\ProgramData\Miniconda3\lib\site-packages\wx\core.py", line 1866, in _Sizer_AddMany
    self.Add(*item)

wxAssertionError: C++ assertion "!(flags & wxALIGN_RIGHT)" failed at ..\..\src\common\sizer.cpp(2168) in wxBoxSizer::DoInsert(): Horizontal alignment flags are ignored in horizontal sizers
```

So I think it's complaining about this line,
`        self.para1.AddMany([(self.para1name,1,wx.EXPAND|wx.ALIGN_LEFT),(self.para1tc,1,wx.EXPAND|wx.ALIGN_RIGHT)])`
are those flag NOT supposed to be there?

**Comment by StephenHLin** [comment]
The issue your seeing is actually not related to python 2 vs 3 directly. It is actually because the latest wx library (used by python3) was updated. Historically, box sizers in wx used to require explicit commands to get the proper look in the UI. 
E.g.:
`self.para1 = wx.BoxSizer(wx.HORIZONTAL)` - line 44
`self.para1.AddMany([(self.para1name,1,wx.EXPAND|wx.ALIGN_LEFT),(self.para1tc,1,wx.EXPAND|wx.ALIGN_RIGHT)])` - line 48

However, in the most up-to-date wx, since `wx.HORIZONTAL` is declared, it is redundant to declare `wx.ALIGN_LEFT` or `wx.ALIGN_RIGHT`, hence the error you are seeing. The wx authors decided to pop an error instead of a warning in the latest version.
Here is a similar situation someone else had: https://forums.wxwidgets.org/viewtopic.php?t=42182

You can simply just delete the ALIGN_LEFT and ALIGN_RIGHT flags and it should resolve the issue.

**Comment by chessybo** [comment]
I got the GUI to come up, fixed the wx issues. I selected the Laser:hp816x and clicked Connect
Output Log:
```
This is pyOptomip version 1.1
Error in sys.excepthook:
Traceback (most recent call last):
  File "C:\ProgramData\Miniconda3\lib\site-packages\IPython\core\interactiveshell.py", line 1826, in showtraceback
    self._showtraceback(etype, value, stb)
  File "C:\ProgramData\Miniconda3\lib\site-packages\ipykernel\zmqshell.py", line 537, in _showtraceback
    sys.stdout.flush()
AttributeError: 'logWriter' object has no attribute 'flush'

Original exception was:
Traceback (most recent call last):
  File "C:\ProgramData\Miniconda3\Lib\site-packages\pyOptomip\hp816x_instrParameters.py", line 61, in connect
    self.laser = hp816x();
  File "C:\ProgramData\Miniconda3\Lib\site-packages\pyOptomip\hp816x_instr.py", line 80, in __init__
    self.hLib = WinDLL('hp816x_32.dll');
  File "C:\ProgramData\Miniconda3\lib\ctypes\__init__.py", line 348, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: [WinError 126] The specified module could not be found
```
I don't really know what I'm doing. For me to do literally anything with this I need hardware? I just thought it was something I could use as a Proof of Concept... 

It was linked to in my edx class so I wanted to explore it.

I can upload what I have if you want to finish where I left off...

**Comment by StephenHLin** [comment]
wx flag update made into new issue to be fixed here: #7

**Comment by StephenHLin** [comment]
Update submitted by chessybo. Needs confirmation testing in lab. #6
